### PR TITLE
Update eks-d release to latest

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: EKSD_LATEST_RELEASES [hunk 1]
```diff
@@ -9,27 +9,27 @@ releases:
...
 - branch: 1-27
-  number: 10
+  number: 11
   kubeVersion: v1.27.4
 - branch: 1-28
-  number: 2
-  kubeVersion: v1.28.0
+  number: 3
+  kubeVersion: v1.28.1
 latest: 1-25
```
[the remark this answers is not in the document]
Comment (jaxesn) [reply EKSD_LATEST_RELEASES:35]: not yet, but thats something we should discuss as to when we want to do that

File: EKSD_LATEST_RELEASES [hunk 2]
```diff
@@ -9,27 +9,27 @@ releases:
   number: 22
   kubeVersion: v1.20.15
 - branch: 1-21
-  number: 26
+  number: 28
```
Comment (abhay-krishna) [inline EKSD_LATEST_RELEASES:12]: we don't need this right?

**Review by jaxesn — APPROVED** [review]
/approve
/unhold

**Pull request merged**